### PR TITLE
fix: add OpenHands filesystem tools to coding agents

### PR DESCRIPTION
**Pull request by kumanday**
## Summary

Configure OpenHands coding-agent conversations with filesystem tools instead of relying on the agent-server `FinishTool`/`ThinkTool` defaults.

## Changes

- add explicit workflow and request-model support for `tools` and `include_default_tools`
- default coding-agent launches to `TerminalTool` and `FileEditorTool`, and persist that selection in the launch profile used for rehydrate/debug flows
- add regression coverage for workflow resolution, request serialization, runner defaults/overrides, and document the new contract in `docs/openhands-agent-server.md`

### Test output

```text
$ cargo test -p opensymphony-workflow
58 passed; 0 failed

$ cargo test -p opensymphony-openhands
61 passed; 0 failed; 2 ignored

$ cargo clippy -p opensymphony-workflow -p opensymphony-openhands --all-targets -- -D warnings
Finished `dev` profile [unoptimized + debuginfo] target(s) successfully
```

### Verification

Confirmed the pre-fix code path could only serialize OpenHands agent `kind` and `llm`, then verified the fixed path now:
- resolves workflow tool overrides and explicit opt-out lists
- defaults coding agents to `TerminalTool` + `FileEditorTool`
- persists tool selection into `launch_profile` so fresh and rehydrated conversations recreate the same request shape

## Checklist

- [x] Tests pass (`cargo test`)
- [x] Code is formatted (`cargo fmt`)
- [x] Clippy is clean (`cargo clippy`)
- [x] Documentation updated (if behavior changed)
- [ ] `AGENTS.md` updated (if repo knowledge changed)
- [x] Evidence section filled out (for substantive changes)

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Performance improvement
- [ ] Refactoring
- [x] Documentation
- [ ] Other: n/a

## Breaking changes

None.

## Related issues

- [COE-293](https://linear.app/trilogy-ai-coe/issue/COE-293/openhands-agent-has-no-filesystem-tools-only-finishtool-and-thinktool)

## Additional notes

- This repo does not have an `elixir/` directory, so the `mix pr_body.check` helper from the generic push skill is not applicable here.

### CODE REVIEW

**Comment by kumanday** [comment]
Reopening to retrigger pull_request checks after no runs were created on PR open.

File: crates/opensymphony-openhands/src/events.rs [hunk 1]
```diff
@@ -395,6 +395,8 @@ mod tests {
                     usage_id: None,
                 },
                 condenser: None,
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-openhands/src/events.rs:397]: Keeping the scope narrow here. `AgentConfig` is a direct OpenHands request model, so I prefer not to add a production `Default` or test-only factory just to hide `None` fields in fixtures. If fixture churn becomes a repeated pain point across follow-up fields, I can address that separately with a targeted test helper refactor.
